include/Fitter/KalmanFitter.hpp

```diff
@@ -229,14 +243,17 @@ class KalmanFitter {
           (result.measurementStates > 0 and state.navigation.navigationBreak)) {
         // printf("Finishing forward filtering");
         result.finished = true;
-        // if (not result.smoothed) {
-        //  printf("Finalize/run smoothing\n");
-        //  auto res = finalize(state, stepper, result);
-        //  if (!res) {
-        //    printf("Error in finalize:\n");
-        //    result.result = false;
-        //  }
-        //}
+        if (not result.smoothed) {
+          
+	  if (IS_MAIN_THREAD) {  
```

Comment:
I guess we need a separate function where only the main thread is used?

Reply:
But here we could remove the `if (IS_MAIN_THREAD) {`, I guess.

Reply:
I added because when the code is ran on CPU, the code is called once for each particle, while called from the GPU, the code is invoked several times. Not sure if that is correct.

Reply:
On the GPU, we could either run with either 1 thread/track or multiple threads/track. They should be implemented with different functions, I guess. The second mode is still under investigation. So in default, we could use just 1 thread/track.